pkg/block/azure/main_test.go

```diff
@@ -0,0 +1,100 @@
+package azure_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+	"github.com/ory/dockertest/v3"
+	"github.com/txn2/txeh"
+)
+
+const (
+	azuriteContainerTimeoutSeconds = 10 * 60 // 10 min
+	containerName                  = "container1"
+	accountName                    = "account1"
+	accountKey                     = "key1"
+)
+
+var (
+	pool     *dockertest.Pool
+	blockURL string
+)
+
+func runAzurite(dockerPool *dockertest.Pool) (string, func()) {
+	ctx := context.Background()
+	resource, err := dockerPool.Run("mcr.microsoft.com/azure-storage/azurite", "latest", []string{
+		fmt.Sprintf("AZURITE_ACCOUNTS=%s:%s", accountName, accountKey),
+	})
+
+	accountHost := fmt.Sprintf("%s.blob.localhost", accountName)
+
+	hosts, err := txeh.NewHostsDefault()
+	if err != nil {
+		panic(err)
+	}
+	hosts.AddHost("127.0.0.1", accountHost)
```

Comment:
Are we overriding the host machine `/etc/hosts` file? If so, I'm worried about this one. is there any other way that doesn't require overriding the machine `/etc/hosts`? Test can crash half way through leaving garbage in that file..

Reply:
Gotcha, so the real problem is that we don't support accounts as part of the path.

Reply:
@nopcoder suggested that we'll leverage `s3.local.lakefs.io` somehow. Is it possible to use a real DNS record that will point to localhost?

Reply:
The domain is comprised from the account name - which is dynamically determined. How do you suggest to deal with that?

Reply:
We add a DNS record `*.local.azurite.lakefs.io` (or something similar) that points to `127.0.0.1`. Then for every account name `xx` we will use the host name `xx.local.azurite.lakefs.io`, thus avoiding hosts file modifications.

Reply:
The usual way is to spin up everything (test and Azurite) inside a single docker-compose environment.

On the one hand, users cannot edit hosts lookup -- this should never work.

In the past we had a DNS record _for ease of deployment_ of local S3 gateways with host-based addressing.  It makes a log of sense for quick-starting users.  But using a DNS record here hurts in other ways, it means "unit" tests require Internet connectivity.

Reply:
As an alternative, you can [override `net.DefaultResolver`](https://go.dev/src/net/lookup.go) with something that resolves `*.azurite.example.net` but falls back on the original DefaultResolver for everything else.